app/src/main/java/com/infomaniak/mail/ui/main/folder/TwoPaneFragment.kt

```diff
@@ -63,46 +71,77 @@ abstract class TwoPaneFragment : Fragment() {
         super.onViewCreated(view, savedInstanceState)
         setupSlidingPane()
         observeFolderChange()
-        observeThreadEvents()
+        observeThreadUid()
+        observeThreadNavigation()
+    }
+
+    override fun onConfigurationChanged(newConfig: Configuration) {
+        super.onConfigurationChanged(newConfig)
+
+        ensureThreadIsOpen()
+
+        val isLeftShown = areBothShown() || isOnlyLeftShown() // TODO: Only works on Phone, and not on Tablet.
+        val statusBarColor = if (isLeftShown && this is ThreadListFragment) {
+            R.color.backgroundHeaderColor
+        } else {
+            R.color.backgroundColor
+        }
+        requireActivity().window.statusBarColor = requireContext().getColor(statusBarColor)
+    }
+
+    override fun onResume() {
+        super.onResume()
+        ensureThreadIsOpen()
     }
 
     private fun setupSlidingPane() {
         slidingPaneLayout.lockMode = SlidingPaneLayout.LOCK_MODE_LOCKED
     }
 
-    private fun observeFolderChange() = with(mainViewModel) {
+    private fun observeFolderChange() = with(twoPaneViewModel) {
 
-        Utils.waitInitMediator(currentFolder, isInSearch).observe(viewLifecycleOwner) { (folder, isInSearch) ->
+        Utils.waitInitMediator(mainViewModel.currentFolder, isInSearch).observe(viewLifecycleOwner) { (folder, isInSearch) ->
 
             val currentFolder = if (isInSearch) {
                 searchFolder
             } else {
                 folder ?: return@observe
             }
 
-            rightPaneFolderName.value = getString(R.string.noConversationSelected, currentFolder.getLocalizedName(context))
+            rightPaneFolderName.value = getString(
+                R.string.noConversationSelected,
+                currentFolder.getLocalizedName(requireContext()),
+            )
 
             if (currentFolder.id != previousFolderId) {
                 previousFolderId = currentFolder.id
                 if (isInThread) closeThread()
             }
+
+            doAfterFolderChanged()
         }
     }
 
-    private fun observeThreadEvents() = with(mainViewModel) {
+    private fun observeThreadUid() {
 
-        val threadListAdapter = when (this@TwoPaneFragment) {
-            is ThreadListFragment -> this@TwoPaneFragment.threadListAdapter
-            is SearchFragment -> this@TwoPaneFragment.threadListAdapter
+        val threadListAdapter = when (this) {
```

Comment:
It is only used in the else, put this there

Reply:
Handled by injection now, much simpler.